### PR TITLE
Fixed FieldDescriptor test

### PR DESCRIPTION
**Pull request by haberman**
Added `TYPE_*` and `LABEL_*` constants to `FieldDescriptor. However we omit the previously-available `CPPTYPE_*` constants, as these are specific to the C++ implementation.

Also added support for finding field descriptors from a fallback database.

### CODE REVIEW

File: python/descriptor.c [hunk 1]
```diff
@@ -1627,5 +1635,30 @@ bool PyUpb_InitDescriptor(PyObject* m) {
...
+      PyUpb_SetIntAttr(field_desc, "TYPE_BYTES", upb_FieldType_Bytes) &&
+      PyUpb_SetIntAttr(field_desc, "TYPE_UINT32", upb_FieldType_UInt32) &&
+      PyUpb_SetIntAttr(field_desc, "TYPE_ENUM", upb_FieldType_Enum) &&
+      PyUpb_SetIntAttr(field_desc, "TYPE_SFIXED32", upb_FieldType_SFixed32) &&
+      PyUpb_SetIntAttr(field_desc, "TYPE_SFIXED64", upb_FieldType_SFixed64) &&
+      PyUpb_SetIntAttr(field_desc, "TYPE_SINT32", upb_FieldType_SInt32) &&
+      PyUpb_SetIntAttr(field_desc, "TYPE_SINT64", upb_FieldType_SInt64) &&
+      PyUpb_SetIntAttr(field_desc, "LABEL_OPTIONAL", kUpb_Label_Optional) &&
+      PyUpb_SetIntAttr(field_desc, "LABEL_REQUIRED", kUpb_Label_Required) &&
+      PyUpb_SetIntAttr(field_desc, "LABEL_REPEATED", kUpb_Label_Repeated);
```
Comment (fowles) [inline python/descriptor.c:1660]: I can't tell if this is in a specific order or not.  In particular, uint32 is not near uint64. Can you simply sort the whole thing?

File: python/descriptor.c [hunk 2]
```diff
@@ -1627,5 +1635,30 @@ bool PyUpb_InitDescriptor(PyObject* m) {
...
+  if (!ok) return false;
+
   return true;
```
Comment (fowles) [inline python/descriptor.c:1663]: pretty sure this is equivalent to `return ok` which means you can drop the local entirely

**Review by fowles — APPROVED** [review]

File: python/descriptor.c (hunk 1 above)
Comment (haberman) [reply python/descriptor.c:1660]: Done. The order is by numeric value, but it's not apparent here so I agree name is nicer.

File: python/descriptor.c (hunk 2 above)
Comment (haberman) [reply python/descriptor.c:1663]: Done.

**Pull request merged**